src/main/java/hudson/tasks/test/AbstractTestResultAction.java

```diff
@@ -355,19 +623,612 @@ private CategoryDataset buildDataSet(StaplerRequest req) {
+        testsToDisplay = Math.min(testsToDisplay, testFailCount.size());
+
+        /*
+        A map storing the testcase along with number of times it flapped and number of times it failed for displaying as
+        tool tip.
+         */
+        flapperInfo = new HashMap<>();
+
+        /*
+        Data structure for indexing all the testcases which will be displayed on the chart and this data structure will
+        also be used for generating tool tips.
+         */
+        mostFailedTestCases = new ArrayList<>(testsToDisplay);
+        xySeries.add(this.run.number + 0.5, testsToDisplay + 0.5);
+        failSeries.add(xySeries);
+        for (int i = 1; i <= testsToDisplay; i++) {
+            xySeries = new XYSeries(i);
+            mostFailedTestCases.add("");
+            failSeries.add(xySeries);
+        }
+
+        /*
+        Populating data structures used by JFreeChart API for displaying trends and generating tool tips.
+         */
+        for (int testIndex = 1; testIndex <= testsToDisplay; testIndex++) {
+            Pair<Integer, String> testCase = testFailCount.get(testIndex - 1);
+            String caseName = testCase.second;
+            int y = testsToDisplay - testIndex + 1;
+            int index = testCaseIndex.get(caseName);
+            List<Integer> infoList = testInfo.get(index);
+            mostFailedTestCases.set(y - 1, caseName);
+            ArrayList<Integer> flapperInfoList = new ArrayList<>();
+            flapperInfoList.add(infoList.get(0));
+            flapperInfoList.add(infoList.get(1));
+            flapperInfo.put(y, flapperInfoList);
+            for (int i = 3; i < infoList.size(); i++) {
+                int x = infoList.get(i);
+                if (i > 3) {
+                    int previousX = infoList.get(i - 1);
+                    if (previousX - x > 1) {
+                        xySeries = failSeries.get(y);
+                        xySeries.add(x + 1, null);
+                    }
+                }
+                xySeries = failSeries.get(y);
+                xySeries.add(x, y);
+            }
+        }
+        LOGGER.log(Level.FINER, "total test trend count for {0}: {1}", new Object[]{run, count});
+
+        /*
+        Building the dataset for displaying charts.
+         */
+        for (int testIndex = testsToDisplay; testIndex >= 0; testIndex--) {
+            dataset.addSeries(failSeries.get(testIndex));
+        }
+        return dataset;
+    }
+
+    /**
+     * A method to get y-axis/range axis label for the trend.
+     * @param req The HTTP request message.
+     * @return Y-axis/range axis label as a string.
+     */
+    private String getYAxisLabel (StaplerRequest req) {
+        String trendType = getParameter(req, TRENDTYPE);
+        if (trendType.equals(BUILD_ANALYSIS) || trendType.equals(LENGTHY_TESTS) || trendType.equals(FLAKY_TESTS))
+            return "Count";
+        else
+            return "count";
```

Comment:
what is with `count/Count` ?

Reply:
As this method was almost useless so, I removed it. Done.